---
title: Different variant callers
icon: post
# This control sidebar order
order: 2
category:
  - Bioinformatics
tag:
  - Tool
---

## SomaticSniper (v1.0.5.0) [^SomaticSniper]

### Installation
```sh
# Build dependencies for Ubuntu
sudo apt-get install build-essential git-core cmake zlib1g-dev libncurses-dev

# Clone the git repo
git clone https://github.com/genome/somatic-sniper.git

# Build SomaticSniper
mkdir somatic-sniper/build
cd somatic-sniper/build
cmake ../
make deps
make -j
make test

# Add path to environment ~/.bashrc
export PATH=$PATH:/PATH_TO/somatic-sniper/build/bin
```
### Usage
```sh
bam-somaticsniper [options] -f <ref.fasta> <tumor.bam> <normal.bam> <snv_output_file>
```
Recommended setting:
```sh
bam-somaticsniper -Q 40 -G -L -f reference.fa tumor.bam normal.bam output.txt
```

## FreeBayes

### Installation
```sh
wget https://github.com/freebayes/freebayes/archive/refs/tags/v1.3.6.zip

unzip v1.3.6.zip
```
### Usage
```sh
freebayes -f [REFERENCE] [OPTIONS] [BAM FILES] >[OUTPUT]
freebayes -f ref.fa aln.bam >var.vcf

freebayes -f ref.fa --pooled-discrete --genotype-qualities tumor.bam normal.bam | vcfsamplediff -s VT normal tumor -

```
Genomic Research paper usage:
```sh
freebayes -f ref.fa  -0 --pooled-discrete --genotype-qualities --min-coverage 20 tumor.bam normal.bam  > output.vcf
```

Then filter the vcf
## vcflib [^vcflib]
### Installation
```sh
conda install -c bioconda vcflib
```
### Usage
Purpose: If multiple allelic primitives (gaps or mismatches) are specified in a single VCF record, split the record into multiple lines, but drop all INFO fields. Does not handle genotypes (yet). MNPs are split into multiple SNPs unless the -m flag is provided. Records generated by splits have th
```sh
vcfallelicprimitives [options] [file]
```

### Installation
```sh
git clone https://github.com/atks/vt.git
cd vt
git submodule update --init --recursive
make
# Test the build
make test
```

## vt [^vt]
### Usage [^vt_usage]
Purpose: Decompose biallelic block substitutions into their constituent SNVs
```sh
#decomposes biallelic block substitutions and write out to decomposed_blocksub.vcf
vt decompose_blocksub gatk.vcf -o decomposed_blocksub.vcf 
```


## Verscan
### Installation
```sh
wget https://github.com/dkoboldt/varscan/releases/download/2.4.2/VarScan.v2.4.2.jar
```
### Usage
```sh
java -jar VarScan.jar  [COMMAND] [OPTIONS]
```

# Prepare fasta to be used as reference genome
```sh
# Create fasta.dict
gatk CreateSequenceDictionary -R ref.fasta

# Create fasta.index
samtools faidx ref.fasta
```

# Samtools
## `samtools view`
```sh
samtools view [options] <in.bam>|<in.sam>|<in.cram> [region ...]
```
Parameters
```js
-S           Ignored (input format is auto-detected)
-b, --bam                  Output BAM
-@, --threads INT
               Number of additional threads to use [0]
-o, --output FILE          Write output to FILE [standard output]
```

## `samtools sort`
Usage:
```sh
samtools sort [options...] [in.bam]
```
Parameters:
```js
-l INT     Set compression level, from 0 (uncompressed) to 9 (best) 
Set the desired compression level for the final output file, ranging from 0 (uncompressed) or 1 (fastest but minimal compression) to 9 (best compression but slowest to write), similarly to gzip(1)'s compression level setting.

-@ INT
Set number of sorting and compression threads. By default, operation is single-threaded.

-o FILE
Write the final sorted output to FILE, rather than to standard output.
```
---------------------
## `samtools fixmate`
Usage:
```sh
samtools fixmate <in.nameSrt.bam> <out.nameSrt.bam>
```
Parameters:
```js
-r           Remove unmapped reads and secondary alignments
-m           Add mate score tag
```

## `samtools markdup`
Usage:
```sh
samtools markdup <input.bam> <output.bam>
```
Parameters:
```js
-r                 Remove duplicate reads
```

[^SomaticSniper]:https://github.com/genome/somatic-sniper
[^vcflib]:https://github.com/vcflib/vcflib
[^vt]:https://github.com/atks/vt
[^vt_usage]:https://genome.sph.umich.edu/wiki/Vt