import{_ as t}from"./_plugin-vue_export-helper.cdc0426e.js";import{o,c as r,b as a,d as e,f as i,e as s,r as p}from"./app.7f4d4b55.js";const l="/assets/WGS_Tumor_merged_sorted_mrkdup_bqsr-scatter.121fd097.png",c={},d=i(`<p>It is easy to find a pipeline and call for results. However, how do you confirm the results are valid? One of the good approaches is to reproduce tutorials.</p><h2 id="goal" tabindex="-1"><a class="header-anchor" href="#goal" aria-hidden="true">#</a> Goal</h2><p>This <sup class="footnote-ref"><a href="#footnote1">[1]</a><a class="footnote-anchor" id="footnote-ref1"></a></sup> uses human genome to identify CNV using cnvkit <sup class="footnote-ref"><a href="#footnote2">[2]</a><a class="footnote-anchor" id="footnote-ref2"></a></sup>. We will reproduce the tutorial&#39;s analysis by downloading their raw data, running the analysis, and comparing the results with their provided outcome to better understand the pipeline.</p><h2 id="download-data" tabindex="-1"><a class="header-anchor" href="#download-data" aria-hidden="true">#</a> Download data</h2><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token function">mkdir</span> data
<span class="token builtin class-name">cd</span> data
<span class="token comment"># Get ref.fa</span>
<span class="token function">wget</span> <span class="token parameter variable">-c</span> http://genomedata.org/pmbio-workshop/references/genome/all/ref_genome.tar
<span class="token function">tar</span> <span class="token parameter variable">-xvf</span> ref_genome.targunzip 
gunzip ref_genome.fa.gz
<span class="token comment"># Get copyCat bed</span>
<span class="token function">wget</span> <span class="token parameter variable">-c</span> http://genomedata.org/pmbio-workshop/misc/copyCat_annotation.zip
<span class="token function">unzip</span> copyCat_annotation.zip 
<span class="token comment"># Get Exom target bed</span>
<span class="token function">wget</span> <span class="token parameter variable">-c</span> http://genomedata.org/pmbio-workshop/references/exome/SeqCapEZ_Exome_v3.0_Design_Annotation_files.zip	
<span class="token function">unzip</span> SeqCapEZ_Exome_v3.0_Design_Annotation_files.zip
<span class="token comment"># Get bam</span>
<span class="token function">wget</span> <span class="token parameter variable">-r</span> <span class="token parameter variable">-np</span> <span class="token parameter variable">-nH</span> --cut-dirs<span class="token operator">=</span><span class="token number">4</span> <span class="token parameter variable">-R</span> <span class="token string">&quot;index.html*&quot;</span> http://genomedata.org/pmbio-workshop/results/chr6_and_chr17/align/
<span class="token comment"># Get wgs bed</span>
<span class="token function">wget</span> http://genomedata.org/pmbio-workshop/results/chr6_and_chr17/somatic/cnvkit_wgs/access-excludes.hg38.chr6_and_17.target.bed
<span class="token comment"># Get results of only chr6 and chr17</span>
<span class="token builtin class-name">cd</span> results
<span class="token function">wget</span> <span class="token parameter variable">-r</span> <span class="token parameter variable">-np</span> <span class="token parameter variable">-nH</span> --cut-dirs<span class="token operator">=</span><span class="token number">4</span> <span class="token parameter variable">-R</span> <span class="token string">&quot;index.html*&quot;</span> http://genomedata.org/pmbio-workshop/results/chr6_and_chr17/somatic/cnvkit_exome/
<span class="token comment"># Get results of wgs</span>
<span class="token function">wget</span> <span class="token parameter variable">-r</span> <span class="token parameter variable">-np</span> <span class="token parameter variable">-nH</span> --cut-dirs<span class="token operator">=</span><span class="token number">4</span> <span class="token parameter variable">-R</span> <span class="token string">&quot;index.html*&quot;</span> http://genomedata.org/pmbio-workshop/results/chr6_and_chr17/somatic/cnvkit_wgs/
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h2 id="run-analysis" tabindex="-1"><a class="header-anchor" href="#run-analysis" aria-hidden="true">#</a> Run analysis</h2><h3 id="prepare-path" tabindex="-1"><a class="header-anchor" href="#prepare-path" aria-hidden="true">#</a> Prepare path</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>conda activate py38_cnvkit
<span class="token builtin class-name">export</span> <span class="token assign-left variable">PYTHONPATH</span><span class="token operator">=</span>PATH_TO_PYTHON
<span class="token comment"># Path</span>
<span class="token assign-left variable">cnvkit_path</span><span class="token operator">=</span><span class="token string">&quot;python -m cnvkit.cnvlib.cnvkit &quot;</span>
<span class="token assign-left variable">WD</span><span class="token operator">=</span>WORK_DIR/
<span class="token assign-left variable">input_path</span><span class="token operator">=</span><span class="token variable">$WD</span>/data
<span class="token assign-left variable">REF</span><span class="token operator">=</span><span class="token variable">$input_path</span>/ref_genome.fa
<span class="token assign-left variable">copyCat_annotation_bed</span><span class="token operator">=</span><span class="token variable">$input_path</span>/copyCat_annotation/gaps.bed
<span class="token assign-left variable">output_path</span><span class="token operator">=</span><span class="token variable">$WD</span>/sandbox

<span class="token function">mkdir</span> <span class="token parameter variable">-p</span> <span class="token variable">$output_path</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h3 id="run-analysis-1" tabindex="-1"><a class="header-anchor" href="#run-analysis-1" aria-hidden="true">#</a> Run analysis</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token comment">###### Prepare data</span>
<span class="token comment"># Calculate the regions of the genome which are inaccessible to sequencing</span>
<span class="token variable">$cnvkit_path</span> access <span class="token variable">$REF</span> <span class="token parameter variable">-x</span> <span class="token variable">$copyCat_annotation_bed</span> <span class="token parameter variable">-o</span> <span class="token variable">$output_path</span>/access-excludes.hg38.bed

<span class="token comment"># cnvkit will complain if access-excludes contains chromosomes not in the bam file</span>
<span class="token comment"># we subset to chr6 and chr17 here to avoid this error later</span>
<span class="token function">grep</span> <span class="token string">&quot;chr6\\|chr17&quot;</span> <span class="token variable">$output_path</span>/access-excludes.hg38.bed <span class="token operator">&gt;</span> <span class="token variable">$output_path</span>/access-excludes.hg38.chr6_and_17.bed

<span class="token comment">####### Run analysis</span>
<span class="token comment"># cnvkit will complain if chromosomes are in the bed file but not the bam, we fix this here</span>
<span class="token function">grep</span> <span class="token string">&quot;chr6\\|chr17&quot;</span> <span class="token variable">$input_path</span>/SeqCapEZ_Exome_v3.0_Design_Annotation_files/SeqCap_EZ_Exome_v3_hg19_primary_targets.bed <span class="token operator">&gt;</span> <span class="token variable">$input_path</span>/SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.chr6_and_17.bed

<span class="token comment"># run the entire cnvkit workflow for the exome data</span>
<span class="token variable">$cnvkit_path</span> batch <span class="token variable">$input_path</span>/Exome_Tumor_sorted_mrkdup_bqsr.bam <span class="token parameter variable">--normal</span> <span class="token variable">$input_path</span>/Exome_Norm_sorted_mrkdup_bqsr.bam <span class="token parameter variable">--targets</span> <span class="token variable">$input_path</span>/SeqCap_EZ_Exome_v3_hg38_primary_targets.v2.chr6_and_17.bed <span class="token parameter variable">--fasta</span> <span class="token variable">$REF</span> <span class="token parameter variable">--access</span> <span class="token variable">$output_path</span>/access-excludes.hg38.chr6_and_17.bed --output-reference <span class="token variable">$output_path</span>/my_reference.cnn --output-dir <span class="token variable">$output_path</span> <span class="token parameter variable">--method</span> hybrid <span class="token parameter variable">-p</span> <span class="token number">8</span> <span class="token parameter variable">--diagram</span> <span class="token parameter variable">--scatter</span> --drop-low-coverage

<span class="token comment">## Run whole genome seq</span>

<span class="token comment"># make directory to store results</span>
<span class="token function">mkdir</span> <span class="token parameter variable">-p</span> <span class="token variable">$output_path</span>/cnvkit_wgs
<span class="token builtin class-name">cd</span> <span class="token variable">$output_path</span>/cnvkit_wgs
<span class="token comment"># run the cnvkit pipeline</span>
<span class="token variable">$cnvkit_path</span> batch <span class="token variable">$input_path</span>/WGS_Tumor_merged_sorted_mrkdup_bqsr.bam <span class="token parameter variable">--normal</span> <span class="token variable">$input_path</span>/WGS_Norm_merged_sorted_mrkdup_bqsr.bam <span class="token parameter variable">--fasta</span> <span class="token variable">$REF</span> <span class="token parameter variable">--access</span> <span class="token variable">$input_path</span>/access-excludes.hg38.chr6_and_17.target.bed --output-reference <span class="token variable">$output_path</span>/cnvkit_wgs/my_reference.cnn --output-dir <span class="token variable">$output_path</span>/cnvkit_wgs <span class="token parameter variable">--method</span> wgs <span class="token parameter variable">-p</span> <span class="token number">8</span> <span class="token parameter variable">--diagram</span> <span class="token parameter variable">--scatter</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h2 id="results" tabindex="-1"><a class="header-anchor" href="#results" aria-hidden="true">#</a> Results</h2><p>There are files and plots results that we can use to compare. And we successfully replicated the tutorial.</p><p><img src="`+l+'" alt="WGS_Tumor_merged_sorted_mrkdup_bqsr-scatter.png" loading="lazy"></p><hr class="footnotes-sep">',14),m={class:"footnotes"},u={class:"footnotes-list"},v={id:"footnote1",class:"footnote-item"},b={href:"https://pmbio.org/module-05-somatic/0005/04/01/Somatic_CNV_Calling/",target:"_blank",rel:"noopener noreferrer"},h=s("https://pmbio.org/module-05-somatic/0005/04/01/Somatic_CNV_Calling/"),k=s(),_=a("a",{href:"#footnote-ref1",class:"footnote-backref"},"\u21A9\uFE0E",-1),g={id:"footnote2",class:"footnote-item"},f={href:"https://github.com/etal/cnvkit",target:"_blank",rel:"noopener noreferrer"},w=s("https://github.com/etal/cnvkit"),x=s(),y=a("a",{href:"#footnote-ref2",class:"footnote-backref"},"\u21A9\uFE0E",-1);function $(q,E){const n=p("ExternalLinkIcon");return o(),r("div",null,[d,a("section",m,[a("ol",u,[a("li",v,[a("p",null,[a("a",b,[h,e(n)]),k,_])]),a("li",g,[a("p",null,[a("a",f,[w,e(n)]),x,y])])])])])}const N=t(c,[["render",$],["__file","CNVkit.html.vue"]]);export{N as default};
