import{_ as t}from"./_plugin-vue_export-helper.cdc0426e.js";import{o,c as i,b as a,d as e,f as r,e as n,r as l}from"./app.a8071ed9.js";const p={},c=r(`<p>Transposable elements (TEs) are mobile DNA sequences capable of replicating themselves within genomes independently of the host cell DNA. They are the &quot;jumping genes&quot; that move from one location on the genome to another. These events will add complexity into large-scale genomic functional analysis. It is crucial to identify TEs and explore the impact of TEs into structural variants identification.</p><h2 id="protocol" tabindex="-1"><a class="header-anchor" href="#protocol" aria-hidden="true">#</a> Protocol</h2><p>According to <sup class="footnote-ref"><a href="#footnote1">[1]</a><a class="footnote-anchor" id="footnote-ref1"></a></sup>, we want to use the tool <code>deTEct</code> <sup class="footnote-ref"><a href="#footnote2">[2]</a><a class="footnote-anchor" id="footnote-ref2"></a></sup> to identify transposition event.</p><h2 id="ingredients" tabindex="-1"><a class="header-anchor" href="#ingredients" aria-hidden="true">#</a> Ingredients</h2><p>Input: Structural variants (VCF file) of PBSV (on PBMM2 alignments) or Sniffles (on NGMLR alignments), transposon annotations (by resonaTE), reference genome (FASTA)</p><h3 id="test" tabindex="-1"><a class="header-anchor" href="#test" aria-hidden="true">#</a> Test</h3><p>I already have SV results called by lumpy, let&#39;s see if it works for this workflow.</p><h2 id="installation" tabindex="-1"><a class="header-anchor" href="#installation" aria-hidden="true">#</a> Installation</h2><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>conda create <span class="token parameter variable">--name</span> py38_deTEct <span class="token assign-left variable">python</span><span class="token operator">=</span><span class="token number">3.8</span>
conda activate py38_deTEct
conda <span class="token function">install</span> <span class="token parameter variable">-c</span> derkevinriehl transposition_detector_detect 
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h2 id="usage" tabindex="-1"><a class="header-anchor" href="#usage" aria-hidden="true">#</a> Usage</h2><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>transposition_deTEct <span class="token parameter variable">-help</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><div class="language-javascript ext-js line-numbers-mode"><pre class="language-javascript"><code>transposition_deTEct <span class="token punctuation">[</span><span class="token operator">-</span>help<span class="token punctuation">]</span> <span class="token operator">-</span>seqHeadTXT <span class="token operator">&lt;</span><span class="token constant">FILE</span><span class="token operator">&gt;</span> <span class="token operator">-</span>transpGFF3 <span class="token operator">&lt;</span><span class="token constant">FILE</span><span class="token operator">&gt;</span> <span class="token operator">-</span>assmFasta <span class="token operator">&lt;</span><span class="token constant">FILE</span><span class="token operator">&gt;</span> <span class="token operator">-</span>svTool <span class="token operator">&lt;</span><span class="token constant">FILE</span><span class="token operator">&gt;</span> <span class="token operator">-</span>svFile <span class="token operator">&lt;</span><span class="token constant">FILE</span><span class="token operator">&gt;</span> <span class="token operator">-</span>outParsedFile <span class="token operator">&lt;</span><span class="token constant">FILE</span><span class="token operator">&gt;</span> <span class="token operator">-</span>outResultFile <span class="token operator">&lt;</span><span class="token constant">FILE</span><span class="token operator">&gt;</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><h2 id="demo" tabindex="-1"><a class="header-anchor" href="#demo" aria-hidden="true">#</a> Demo</h2><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token function">git</span> clone https://github.com/DerKevinRiehl/transposition_detector_deTEct.git
<span class="token builtin class-name">cd</span> PATH_TO/transposition_detector_deTEct
conda activate py38_deTEct
<span class="token comment"># demo for sniffles_ngmlr alignments</span>
transposition_deTEct <span class="token parameter variable">-seqHeadTXT</span> demoFiles/sequence_heads.txt <span class="token parameter variable">-transpGFF3</span> demoFiles/FinalAnnotations_Transposons.gff3 <span class="token parameter variable">-assmFasta</span> demoFiles/sequence_CB4856.fasta <span class="token parameter variable">-svTool</span> sniffles <span class="token parameter variable">-svFile</span> demoFiles/SX3351_addisababa.sniffles_ngmlr.vcf <span class="token parameter variable">-outParsedFile</span> demoFiles/sniffles_ngmlr/SX3351_addisababa.SV.vcf.gff3 <span class="token parameter variable">-outResultFile</span> demoFiles/sniffles_ngmlr/SX3351_addisababa.transpositionEvents.gff3

<span class="token comment"># demo for pbsv_pbsmm2 alignments</span>
transposition_deTEct <span class="token parameter variable">-seqHeadTXT</span> demoFiles/sequence_heads.txt <span class="token parameter variable">-transpGFF3</span> demoFiles/FinalAnnotations_Transposons.gff3 <span class="token parameter variable">-assmFasta</span> demoFiles/sequence_CB4856.fasta <span class="token parameter variable">-svTool</span> pbsv <span class="token parameter variable">-svFile</span> demoFiles/SX3351_addisababa.pbsv_pbmm2.vcf <span class="token parameter variable">-outParsedFile</span> demoFiles/pbsv_pbmm2/SX3351_addisababa.SV.vcf.gff3 <span class="token parameter variable">-outResultFile</span> demoFiles/pbsv_pbmm2/SX3351_addisababa.transpositionEvents.gff3
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>The actual scripts</p><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>conda activate py38_deTEct
<span class="token assign-left variable">seqhead</span><span class="token operator">=</span>/home/ybao2/GithubWorkRepo/BulkDNA/transposon_annotation_reasonaTE/workspace/testProject/sequence_heads.txt
<span class="token assign-left variable">TE</span><span class="token operator">=</span>/home/ybao2/GithubWorkRepo/BulkDNA/transposon_annotation_reasonaTE/workspace/testProject/finalResults/FinalAnnotations_Transposons.gff3
<span class="token assign-left variable">genome</span><span class="token operator">=</span>/media/XLStorage/ybao2/RefGenome/dmel-all-chromosome-r6.39.fasta
<span class="token assign-left variable">SV</span><span class="token operator">=</span>demoFiles/SX3351_addisababa.sniffles_ngmlr.vcf
<span class="token assign-left variable">outparse</span><span class="token operator">=</span>demoFiles/sniffles_ngmlr/SX3351_addisababa.SV.vcf.gff3
<span class="token assign-left variable">outresult</span><span class="token operator">=</span>demoFiles/sniffles_ngmlr/SX3351_addisababa.transpositionEvents.gff3

transposition_deTEct <span class="token parameter variable">-seqHeadTXT</span> <span class="token variable">$seqhead</span> <span class="token parameter variable">-transpGFF3</span> <span class="token variable">$TE</span> <span class="token parameter variable">-assmFasta</span> <span class="token variable">$genome</span> <span class="token parameter variable">-svTool</span> sniffles <span class="token parameter variable">-svFile</span> <span class="token variable">$SV</span> <span class="token parameter variable">-outParsedFile</span> <span class="token variable">$outparse</span> <span class="token parameter variable">-outResultFile</span> <span class="token variable">$outresult</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>which tells us we need multiple files produced by the tool <code>reasonaTE</code> <sup class="footnote-ref"><a href="#footnote3">[3]</a><a class="footnote-anchor" id="footnote-ref3"></a></sup></p><h2 id="reasonate" tabindex="-1"><a class="header-anchor" href="#reasonate" aria-hidden="true">#</a> <code>reasonaTE</code></h2><h3 id="ingredients-1" tabindex="-1"><a class="header-anchor" href="#ingredients-1" aria-hidden="true">#</a> Ingredients</h3><ul><li>Input: Genome assembly (FASTA file).</li><li>Output: Lots of transposon annotations (GFF3 file).</li></ul><h3 id="installation-1" tabindex="-1"><a class="header-anchor" href="#installation-1" aria-hidden="true">#</a> Installation</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token comment"># Environment 1 - including all annotation tools</span>
conda create <span class="token parameter variable">-y</span> <span class="token parameter variable">--name</span> transposon_annotation_tools_env <span class="token assign-left variable">python</span><span class="token operator">=</span><span class="token number">2.7</span>
conda activate transposon_annotation_tools_env
conda <span class="token function">install</span> <span class="token parameter variable">-y</span> mamba
conda <span class="token function">install</span> <span class="token parameter variable">-c</span> bioconda repeatmodeler repeatmasker <span class="token comment"># Recommended not too install via conda</span>
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> bioconda genometools-genometools <span class="token comment"># for some users: mamba install -y -c bioconda -c conda-forge genometools-genometools</span>
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> derkevinriehl transposon_annotation_reasonate
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> derkevinriehl transposon_annotation_tools_proteinncbicdd1000
conda <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> derkevinriehl transposon_annotation_tools_transposonpsicli
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> derkevinriehl transposon_annotation_tools_mitetracker
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> derkevinriehl <span class="token assign-left variable">transposon_annotation_tools_sinescan</span><span class="token operator">=</span><span class="token number">1.1</span>.2
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> derkevinriehl transposon_annotation_tools_helitronscanner
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> derkevinriehl transposon_annotation_tools_mitefinderii
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> derkevinriehl transposon_annotation_tools_mustv2
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> derkevinriehl transposon_annotation_tools_sinefinder
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> anaconda biopython
conda deactivate
<span class="token comment"># Environment 2 - including CD-Hit and Transposon Classifier RFSB</span>
conda create <span class="token parameter variable">-y</span> <span class="token parameter variable">--name</span> transposon_annotation_reasonaTE
conda activate transposon_annotation_reasonaTE
conda <span class="token function">install</span> <span class="token parameter variable">-y</span> mamba
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> anaconda biopython
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> bioconda cd-hit blast seqkit
mamba <span class="token function">install</span> <span class="token parameter variable">-y</span> <span class="token parameter variable">-c</span> derkevinriehl transposon_annotation_reasonate transposon_classifier_rfsb
conda deactivate
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>Issue: The process is very slow and <code>mamba</code> is not working with error message:</p><div class="language-javascript ext-js line-numbers-mode"><pre class="language-javascript"><code><span class="token operator">/</span>anaconda3<span class="token operator">/</span>envs<span class="token operator">/</span>transposon_annotation_tools_env<span class="token operator">/</span>lib<span class="token operator">/</span>python2<span class="token punctuation">.</span><span class="token number">7</span><span class="token operator">/</span>site<span class="token operator">-</span>packages<span class="token operator">/</span>tqdm<span class="token operator">/</span>__init__<span class="token punctuation">.</span>py&quot;<span class="token punctuation">,</span> line <span class="token number">3</span><span class="token punctuation">,</span> <span class="token keyword">in</span> <span class="token operator">&lt;</span>module<span class="token operator">&gt;</span>
    from <span class="token punctuation">.</span>cli <span class="token keyword">import</span> main  # <span class="token constant">TODO</span><span class="token operator">:</span> remove <span class="token keyword">in</span> v5<span class="token punctuation">.</span><span class="token number">0.0</span>
  File <span class="token string">&quot;/home/ybao2/anaconda3/envs/transposon_annotation_tools_env/lib/python2.7/site-packages/tqdm/cli.py&quot;</span><span class="token punctuation">,</span> line <span class="token number">202</span>
    sys<span class="token punctuation">.</span>stderr<span class="token punctuation">.</span><span class="token function">write</span><span class="token punctuation">(</span>f<span class="token string">&quot;Error:Unknown argument:{argv[0]}\\n{help_short}&quot;</span><span class="token punctuation">)</span>
                                                                     <span class="token operator">^</span>
<span class="token literal-property property">SyntaxError</span><span class="token operator">:</span> invalid syntax
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h3 id="new-attempt" tabindex="-1"><a class="header-anchor" href="#new-attempt" aria-hidden="true">#</a> New attempt</h3><p>Remove the conda env</p><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token function">wget</span> https://raw.githubusercontent.com/DerKevinRiehl/transposon_annotation_reasonaTE/main/environment_yml/transposon_annotation_tools_env.yml
<span class="token function">wget</span> https://raw.githubusercontent.com/DerKevinRiehl/transposon_annotation_reasonaTE/main/environment_yml/transposon_annotation_reasonaTE.yml
conda <span class="token function">env</span> create <span class="token parameter variable">-f</span> transposon_annotation_tools_env.yml
conda <span class="token function">env</span> create <span class="token parameter variable">-f</span> transposon_annotation_reasonaTE.yml
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>It seems to work.</p><h3 id="usage-1" tabindex="-1"><a class="header-anchor" href="#usage-1" aria-hidden="true">#</a> Usage</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>conda activate transposon_annotation_tools_env
<span class="token function">mkdir</span> workspace
<span class="token function">wget</span> https://raw.githubusercontent.com/DerKevinRiehl/transposon_annotation_reasonaTE/main/workspace/testProject/sequence.fasta <span class="token comment"># demo fasta you could use</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>The actual script</p><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token shebang important">#!/bin/sh</span>
<span class="token comment">#SBATCH --qos=normal            # Quality of Service</span>
<span class="token comment">#SBATCH --job-name=reasonaTE     # Job Name</span>
<span class="token comment">#SBATCH --time=1-0:00:00         # WallTime</span>
<span class="token comment">#SBATCH --nodes=1               # Number of Nodes</span>
<span class="token comment">#SBATCH --ntasks-per-node=1     # Number of tasks (MPI processes)</span>
<span class="token comment">#SBATCH --cpus-per-task=1	# Number of threads per task (OMP threads)</span>
<span class="token comment">#SBATCH --output=Log/reasonaTE.out	  ### File in which to store job output</span>

<span class="token comment"># Activate annaconda to make it usable in slurm</span>
<span class="token builtin class-name">eval</span> <span class="token string">&quot;<span class="token variable"><span class="token variable">$(</span>conda shell.bash hook<span class="token variable">)</span></span>&quot;</span>

<span class="token assign-left variable">workspace</span><span class="token operator">=</span>workspace
<span class="token assign-left variable">projectname</span><span class="token operator">=</span>testProject
<span class="token assign-left variable">genome</span><span class="token operator">=</span>sequence.fasta
<span class="token function">mkdir</span> <span class="token parameter variable">-p</span> <span class="token variable">$workspace</span>

<span class="token comment"># 1. Create a project</span>
conda activate transposon_annotation_tools_env
reasonaTE <span class="token parameter variable">-mode</span> createProject <span class="token parameter variable">-projectFolder</span> <span class="token variable">$workspace</span> <span class="token parameter variable">-projectName</span> <span class="token variable">$projectname</span> <span class="token parameter variable">-inputFasta</span> <span class="token variable">$genome</span>
<span class="token comment"># 2. Annotate genome with annotation tools</span>
conda activate transposon_annotation_tools_env
reasonaTE <span class="token parameter variable">-mode</span> annotate <span class="token parameter variable">-projectFolder</span> <span class="token variable">$workspace</span> <span class="token parameter variable">-projectName</span> <span class="token variable">$projectname</span> <span class="token parameter variable">-tool</span> all
<span class="token comment"># 3. Parse annotations</span>
conda activate transposon_annotation_tools_env
reasonaTE <span class="token parameter variable">-mode</span> parseAnnotations <span class="token parameter variable">-projectFolder</span> <span class="token variable">$workspace</span> <span class="token parameter variable">-projectName</span> <span class="token variable">$projectname</span>
<span class="token comment"># 4. Run the pipeline on the genome annotations</span>
conda activate transposon_annotation_reasonaTE
reasonaTE <span class="token parameter variable">-mode</span> pipeline <span class="token parameter variable">-projectFolder</span> <span class="token variable">$workspace</span> <span class="token parameter variable">-projectName</span> <span class="token variable">$projectname</span>
<span class="token comment"># 5. Calculate final statistics</span>
conda activate transposon_annotation_reasonaTE
reasonaTE <span class="token parameter variable">-mode</span> statistics <span class="token parameter variable">-projectFolder</span> <span class="token variable">$workspace</span> <span class="token parameter variable">-projectName</span> <span class="token variable">$projectname</span>

</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h3 id="check" tabindex="-1"><a class="header-anchor" href="#check" aria-hidden="true">#</a> Check!</h3><h4 id="step-3-annotate-results" tabindex="-1"><a class="header-anchor" href="#step-3-annotate-results" aria-hidden="true">#</a> Step 3: Annotate results</h4><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>conda activate transposon_annotation_tools_env
reasonaTE <span class="token parameter variable">-mode</span> checkAnnotations <span class="token parameter variable">-projectFolder</span> workspace <span class="token parameter variable">-projectName</span> testProject
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div></div></div><pre>reasonaTE -mode checkAnnotations -projectFolder workspace -projectName testProject
Checking helitronScanner        ... completed
Checking ltrHarvest     ... completed
Checking ltrPred        ... not completed
Checking mitefind       ... completed
Checking mitetracker    ... completed
Checking must   ... completed
Checking repeatmodel    ... not completed
Checking repMasker      ... not completed
Checking sinefind       ... completed
Checking sinescan       ... completed
Checking tirvish        ... completed
Checking transposonPSI  ... completed
Checking NCBICDD1000    ... completed
</pre><div class="custom-container tip"><p class="custom-container-title">Notice there is no result for \`ItrPred\`</p><p>To run <code>ItrPred</code>, do separate installation following the tutorial <sup class="footnote-ref"><a href="#footnote4">[4]</a><a class="footnote-anchor" id="footnote-ref4"></a></sup></p></div><div class="custom-container tip"><p class="custom-container-title">Notice there is no result for \`repeatmodel\` and \`repMasker\`</p><p>Previously, I tried to use conda to work with these two and failed. The conda environment for these two tools have been reported errors. Running with the docker image also got me nowhere. I need to revisit this bit to install from source.</p></div><h4 id="step-3-parse-annotations" tabindex="-1"><a class="header-anchor" href="#step-3-parse-annotations" aria-hidden="true">#</a> Step 3: Parse annotations:</h4><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>conda activate transposon_annotation_tools_env
reasonaTE <span class="token parameter variable">-mode</span> checkParsed <span class="token parameter variable">-projectFolder</span> workspace <span class="token parameter variable">-projectName</span> testProject
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div></div></div><pre>29       NCBICDD1000.gff3
9        helitronScanner.gff3
180      ltrHarvest.gff3
0        mitefind.gff3
41       mitetracker.gff3
25       must.gff3
12       sinefind.gff3
90       sinescan.gff3
295      tirvish.gff3
15       transposonPSI.gff3
</pre><h3 id="supplementary" tabindex="-1"><a class="header-anchor" href="#supplementary" aria-hidden="true">#</a> Supplementary</h3><p>Annotation tools list</p><div class="language-javascript ext-js line-numbers-mode"><pre class="language-javascript"><code>helitronScanner
ltrHarvest
mitefind
mitetracker
must
repeatmodel
repMasker
sinefind
sinescan
tirvish
transposonPSI
<span class="token constant">NCBICDD1000</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><hr class="footnotes-sep">`,45),d={class:"footnotes"},m={class:"footnotes-list"},v={id:"footnote1",class:"footnote-item"},b={href:"https://academic.oup.com/nar/article/50/11/e64/6541023",target:"_blank",rel:"noopener noreferrer"},u=n("https://academic.oup.com/nar/article/50/11/e64/6541023"),k=n(),h=a("a",{href:"#footnote-ref1",class:"footnote-backref"},"\u21A9\uFE0E",-1),f={id:"footnote2",class:"footnote-item"},_={href:"https://github.com/DerKevinRiehl/transposition_detector_deTEct",target:"_blank",rel:"noopener noreferrer"},g=n("https://github.com/DerKevinRiehl/transposition_detector_deTEct"),T=n(),y=a("a",{href:"#footnote-ref2",class:"footnote-backref"},"\u21A9\uFE0E",-1),E={id:"footnote3",class:"footnote-item"},F={href:"https://github.com/DerKevinRiehl/transposon_annotation_reasonaTE",target:"_blank",rel:"noopener noreferrer"},w=n("https://github.com/DerKevinRiehl/transposon_annotation_reasonaTE"),j=n(),x=a("a",{href:"#footnote-ref3",class:"footnote-backref"},"\u21A9\uFE0E",-1),S={id:"footnote4",class:"footnote-item"},C={href:"https://github.com/DerKevinRiehl/transposon_annotation_reasonaTE/blob/main/TutorialRunLTRPred.md",target:"_blank",rel:"noopener noreferrer"},I=n("https://github.com/DerKevinRiehl/transposon_annotation_reasonaTE/blob/main/TutorialRunLTRPred.md"),P=n(),N=a("a",{href:"#footnote-ref4",class:"footnote-backref"},"\u21A9\uFE0E",-1);function R(q,A){const s=l("ExternalLinkIcon");return o(),i("div",null,[c,a("section",d,[a("ol",m,[a("li",v,[a("p",null,[a("a",b,[u,e(s)]),k,h])]),a("li",f,[a("p",null,[a("a",_,[g,e(s)]),T,y])]),a("li",E,[a("p",null,[a("a",F,[w,e(s)]),j,x])]),a("li",S,[a("p",null,[a("a",C,[I,e(s)]),P,N])])])])])}const $=t(p,[["render",R],["__file","deTEct.html.vue"]]);export{$ as default};
