const t=JSON.parse(`{"key":"v-5f9ac6b4","path":"/blog/bioinformatics/GATK_SNV.html","title":"GATK Somatic Short Variant Discovery (SNVs and Indels)","lang":"en-US","frontmatter":{"title":"GATK Somatic Short Variant Discovery (SNVs and Indels)","icon":"post","order":2,"category":["Bioinformatics"],"tag":["Tool","GATK"],"summary":"Here are some notes about calling somatic mutations using GATK mutect2. Reference - GATK tutorials: GATK Somatic short variant discovery (SNVs + Indels) Version I Version II Relate","head":[["meta",{"property":"og:url","content":"https://vuepress-theme-hope-v2-demo.mrhope.site/blog/bioinformatics/GATK_SNV.html"}],["meta",{"property":"og:site_name","content":"Yuwei Bao's personal website"}],["meta",{"property":"og:title","content":"GATK Somatic Short Variant Discovery (SNVs and Indels)"}],["meta",{"property":"og:type","content":"article"}],["meta",{"property":"og:updated_time","content":"2022-11-22T04:24:11.000Z"}],["meta",{"property":"og:locale","content":"en-US"}],["meta",{"property":"article:tag","content":"Tool"}],["meta",{"property":"article:tag","content":"GATK"}],["meta",{"property":"article:modified_time","content":"2022-11-22T04:24:11.000Z"}]]},"excerpt":"","headers":[{"level":2,"title":"Reference - GATK tutorials:","slug":"reference-gatk-tutorials","link":"#reference-gatk-tutorials","children":[]},{"level":2,"title":"1. Call somatic short variants and generate a bamout with Mutect2","slug":"_1-call-somatic-short-variants-and-generate-a-bamout-with-mutect2","link":"#_1-call-somatic-short-variants-and-generate-a-bamout-with-mutect2","children":[]},{"level":2,"title":"2. Create a sites-only PoN with CreateSomaticPanelOfNormals","slug":"_2-create-a-sites-only-pon-with-createsomaticpanelofnormals","link":"#_2-create-a-sites-only-pon-with-createsomaticpanelofnormals","children":[]},{"level":2,"title":"3. Estimate cross-sample contamination using GetPileupSummaries and CalculateContamination.","slug":"_3-estimate-cross-sample-contamination-using-getpileupsummaries-and-calculatecontamination","link":"#_3-estimate-cross-sample-contamination-using-getpileupsummaries-and-calculatecontamination","children":[{"level":3,"title":"What if I find high levels of contamination?","slug":"what-if-i-find-high-levels-of-contamination","link":"#what-if-i-find-high-levels-of-contamination","children":[]}]},{"level":2,"title":"4. Filter for confident somatic calls using FilterMutectCalls","slug":"_4-filter-for-confident-somatic-calls-using-filtermutectcalls","link":"#_4-filter-for-confident-somatic-calls-using-filtermutectcalls","children":[]},{"level":2,"title":"5. (Optional) Estimate artifacts with CollectSequencingArtifactMetrics and filter them with FilterByOrientationBias","slug":"_5-optional-estimate-artifacts-with-collectsequencingartifactmetrics-and-filter-them-with-filterbyorientationbias","link":"#_5-optional-estimate-artifacts-with-collectsequencingartifactmetrics-and-filter-them-with-filterbyorientationbias","children":[]},{"level":2,"title":"6. Set up in IGV to review somatic calls","slug":"_6-set-up-in-igv-to-review-somatic-calls","link":"#_6-set-up-in-igv-to-review-somatic-calls","children":[]},{"level":2,"title":"Another tutorial","slug":"another-tutorial","link":"#another-tutorial","children":[]}],"git":{"createdTime":1664689987000,"updatedTime":1669091051000,"contributors":[{"name":"yuweibao15","email":"ybao2@tulane.edu","commits":3}]},"readingTime":{"minutes":2.48,"words":743},"filePathRelative":"blog/bioinformatics/GATK_SNV.md","localizedDate":"October 2, 2022"}`);export{t as data};
