import{_ as s}from"./_plugin-vue_export-helper.cdc0426e.js";import{o as a,c as r,b as e,d as n,f as i,e as o,r as l}from"./app.adc61bf5.js";const h={},c=i('<h2 id="must-read-papers-for-graduate-students" tabindex="-1"><a class="header-anchor" href="#must-read-papers-for-graduate-students" aria-hidden="true">#</a> Must read papers for graduate students</h2><p><strong>This list was organized and posted on <sup class="footnote-ref"><a href="#footnote1">[1]</a><a class="footnote-anchor" id="footnote-ref1"></a></sup> by Bob Thomson <sup class="footnote-ref"><a href="#footnote2">[2]</a><a class="footnote-anchor" id="footnote-ref2"></a></sup> in 2015</strong>:</p><p>Bull, J. J., Huelsenbeck, J. P., Cunningham, C. W., Swofford, D. L., &amp; Waddel, P. J. (1993). Partitioning and combining data in phylogenetic analysis. Systematic Biology, 42(3), 384\u2013397.</p><p>Cavalli-Sforza, L. L., &amp; Edwards, a W. F. (1967). Phylogenetic analysis. Models and estimation procedures. The American Journal of Human Genetics, 19, 233\u2013257.</p><p>Edwards, S. V. (2009). Is a new and general theory of molecular systematics emerging? Evolution, 63, 1\u201319.</p><p>Felsenstein, J. (1973). Maximum likelihood and minimum-steps methods for estimating evolutionary trees from data on discrete characters. Systematic Biology, 22, 240\u2013249.</p><p>Felsenstein, J. (1978). Cases in which parsimony or compatibility methods will be positively misleading. Systematic Zoology, 27, 401\u2013410.</p><p>Felsenstein, J. (1981). Evolutionary trees from DNA sequences: a maximum likelihood approach. Journal of Molecular Evolution, 17, 368\u2013376.</p><p>Felsenstein, J. (1985). Confidence limits on phylogenies: an approach using the bootstrap. Evolution, 39, 783\u2013791.</p><p>Felsenstein, J. (1985). Phylogenies and the comparative method. American Naturalist, 125, 1\u201315.</p><p>Goldman, N. (1993). Statistical tests of models of DNA substitution. Journal of Molecular Evolution, 36, 182\u2013198.</p><p>Hillis, D. M., &amp; Bull, J. J. (1993). An Empirical Test of Bootstrapping as a Method for Assessing Confidence in Phylogenetic Analysis. Systematic Biology, 42, 182\u2013192.</p><p>Holder, M., &amp; Lewis, P. O. (2003). Phylogeny estimation: traditional and Bayesian approaches. Nature Reviews. Genetics, 4, 275\u2013284.</p><p>Kumar, S., Filipski, A. J., Battistuzzi, F. U., Kosakovsky Pond, S. L., &amp; Tamura, K. (2012). Statistics and truth in phylogenomics. Molecular Biology and Evolution, 29, 457\u2013472.</p><p>Maddison, W. P. (1997). Gene Trees in Species Trees. Systematic Biology, 46, 523\u2013536.</p><p>Pauling, L., &amp; Zuckerkandl, E. (1963). Chemical paleogenetics. Acta Chem. Scand, 17, S9 \u2013 S16.</p><p>Sullivan, J., &amp; Swofford, D. (1997). Are Guinea Pigs Rodents?? The Importance of Adequate Models in Molecular Phylogenetics. Journal of Mammalian Evolution, 4, 77\u201386.</p><h2 id="papers-related-to-hmm" tabindex="-1"><a class="header-anchor" href="#papers-related-to-hmm" aria-hidden="true">#</a> Papers related to HMM</h2><p><strong>The following were recommended by by Joe Felsenstein <sup class="footnote-ref"><a href="#footnote3">[3]</a><a class="footnote-anchor" id="footnote-ref3"></a></sup> in 2016:</strong></p><p>Yang, Z. 1994. Maximum-likelihood estimation of phylogeny from DNA sequences when substitution rates differ over sites. Molecular Biology and Evolution 10: 1396-1401. [Use of gamma distribution of rate variation in ML phylogenies]</p><p>Yang, Z. 1994. Maximum likelihood phylogenetic estimation from DNA sequences with variable rates over sites: approximate methods. Journal of Molecular Evolution 39: 306-314. [Approximating gamma distribution in ML phylogenies by an HMM]</p><p>Yang, Z. 1995. A space-time process model for the evolution of DNA sequences. Genetics 139: 993-1005. [Allowing for autocorrelated rates along the molecule using an HMM for ML phylogenies]</p><p>Felsenstein, J. and G. A. Churchill. 1996. A Hidden Markov Model approach to variation among sites in rate of evolution Molecular Biology and Evolution 13: 93-104. [HMM approach to evolutionary rate variation]</p><p>Thorne, J. L., N. Goldman, and D. T. Jones. 1996. Combining protein evolution and secondary structure. Molecular Biology and Evolution 13 666-673. [HMM for secondary structure of proteins, with phylogenies]</p><h2 id="papers-related-to-bayesian-model-comparison" tabindex="-1"><a class="header-anchor" href="#papers-related-to-bayesian-model-comparison" aria-hidden="true">#</a> Papers related to Bayesian model comparison</h2><p><strong>The following were recommended by Jeffrey Thorne <sup class="footnote-ref"><a href="#footnote4">[4]</a><a class="footnote-anchor" id="footnote-ref4"></a></sup> in 2016:</strong></p><p>Posterior Predictive Inference in Phylogenetics: J.P. Bollback. 2002. Molecular Biology and Evolution. 19:1171-1180</p><p>Harmonic Mean and other techniques for estimating Bayes factors: Newton and Raftery. 1994. Journal of the Royal Statistical Society. Series B. 56(1):3-48.</p><div class="custom-container tip"><p class="custom-container-title">More reliable ways to approximate marginal likelihood</p><p>Thermodynamic Integration to Approximate Bayes Factors (adapted to molecular evolution data): Lartillot and Philippe. 2006. Syst. Biol. 55:195-207</p><p>Improving marginal likelihood estimation for Bayesian phylogenetic model selection. W. Xie, P.O. Lewis, Y. Fan, L. Kao, M-H Chen. 2011. Syst Biol. 60(2):150-160.</p><p>Choosing among partition models in Bayesian phylogenetics. Y. Fan, R. Wu, M-H Chen, L Kuo, P.O. Lewis. 2011. Mol. Biol. Evol. 28(1):523-532.</p></div><p>Markov chain Monte Carlo without likelihoods. P. Marjoram, J. Molitor, V. Plagnol, and S. Tavare. 2003. PNAS USA. 100(26): 15324-15328.</p><p>H. Jeffreys. The Theory of Probability (3e). Oxford (1961); p. 432</p><p>M.A. Beaumont, W. Zhang, D.J. Balding. Approximate Bayesian Computation in Population Genetics. 2002. Genetics 162:2025-2035.</p><h2 id="papers-related-to-markov-chain-monte-carlo" tabindex="-1"><a class="header-anchor" href="#papers-related-to-markov-chain-monte-carlo" aria-hidden="true">#</a> Papers related to Markov Chain Monte Carlo</h2><p><strong>The following were recommended by Jeffery Thorne <sup class="footnote-ref"><a href="#footnote5">[5]</a><a class="footnote-anchor" id="footnote-ref5"></a></sup></strong></p><p>CARLIN, B.P., and T.A. LOUIS. 1996. Bayes and Empirical Bayes Methods for Data Analysis. Chapman and Hall, London.</p><p>GELMAN, A., J.B. CARLIN, H.S. STERN, and D.B. RUBIN. 1995. Bayesian Data Analysis. Chapman and Hall, London.</p><p>Hastings WK (1970) Monte Carlo sampling methods using Markov chains and their applications. Biometrika 57:97\u2013109</p><p>METROPOLIS, N., A.W. ROSENBLUTH, M.N. ROSENBLUTH, A.H. TELLER, and E. TELLER. 1953. Equations of state calculations by fast computing machines. J. Chem. Phys. 21: 1087\u20131092.</p><p><strong>The following were recommended by Matthew Stephens and Eric C. Anderson</strong></p><h3 id="examples-of-importance-sampling-in-genetics" tabindex="-1"><a class="header-anchor" href="#examples-of-importance-sampling-in-genetics" aria-hidden="true">#</a> Examples of importance sampling in genetics:</h3>',40),d={href:"https://pubmed.ncbi.nlm.nih.gov/7800710/",target:"_blank",rel:"noopener noreferrer"},p=o("Griffiths and Tavare (1994)"),u={href:"https://doi.org/10.1111/1467-9868.00254",target:"_blank",rel:"noopener noreferrer"},f=o("Matt Stephens and Peter Donnelly (2000)"),m={href:"https://doi.org/10.1534/genetics.105.048074",target:"_blank",rel:"noopener noreferrer"},_=o("Anderson and Garza (2006)"),g=e("h3",{id:"papers-related-to",tabindex:"-1"},[e("a",{class:"header-anchor",href:"#papers-related-to","aria-hidden":"true"},"#"),o(" Papers related to "),e("span",{class:"katex"},[e("span",{class:"katex-mathml"},[e("math",{xmlns:"http://www.w3.org/1998/Math/MathML"},[e("semantics",null,[e("mrow",null,[e("mi",null,"M"),e("msup",null,[e("mi",null,"C"),e("mn",null,"3")])]),e("annotation",{encoding:"application/x-tex"},"MC^3")])])]),e("span",{class:"katex-html","aria-hidden":"true"},[e("span",{class:"base"},[e("span",{class:"strut",style:{height:"0.8141em"}}),e("span",{class:"mord mathnormal",style:{"margin-right":"0.10903em"}},"M"),e("span",{class:"mord"},[e("span",{class:"mord mathnormal",style:{"margin-right":"0.07153em"}},"C"),e("span",{class:"msupsub"},[e("span",{class:"vlist-t"},[e("span",{class:"vlist-r"},[e("span",{class:"vlist",style:{height:"0.8141em"}},[e("span",{style:{top:"-3.063em","margin-right":"0.05em"}},[e("span",{class:"pstrut",style:{height:"2.7em"}}),e("span",{class:"sizing reset-size6 size3 mtight"},[e("span",{class:"mord mtight"},"3")])])])])])])])])])])],-1),y={href:"https://www.jstor.org/stable/2246094",target:"_blank",rel:"noopener noreferrer"},b=o("Metropolis coupled MCMC (Geyer 1991)"),M=e("h2",{id:"other-awesome-recourses",tabindex:"-1"},[e("a",{class:"header-anchor",href:"#other-awesome-recourses","aria-hidden":"true"},"#"),o(" Other awesome recourses")],-1),k={href:"https://felsenst.github.io/courses.html",target:"_blank",rel:"noopener noreferrer"},v=o("Joe Felsenstein's courses"),w={href:"https://evolution.gs.washington.edu/sisg/2016/",target:"_blank",rel:"noopener noreferrer"},B=o("2016 SISG Module 19: Molecular Phylogenetics (Instructors: Mark Holder, Jeffery Thorne, and Joe Felsenstein)"),S={href:"http://www2.stat.duke.edu/~rcs46/bayes17.html",target:"_blank",rel:"noopener noreferrer"},C=o("Bayesian Methods Class by Rebecca C. Steorts at Duke University"),E={href:"http://treethinkers.org/2019-bodega-applied-phylogenetics-workshop-schedule/",target:"_blank",rel:"noopener noreferrer"},A=o("2019 Bodega Applied Phylogenetics Workshop by UC Davis & Bodega Marine Laboratory"),P={href:"http://phyloseminar.org/",target:"_blank",rel:"noopener noreferrer"},x=o('Phylogenetics Seminars organized by Frederick "Erick" Matsen'),L={href:"http://www.phylobabble.org/",target:"_blank",rel:"noopener noreferrer"},J=o('Phylogenetics discussion forum organized by Frederick "Erick" Matsen'),T={href:"https://plewis.github.io/phylogenetics2022/",target:"_blank",rel:"noopener noreferrer"},N=o("Paul O. Lewis' grad-level Phylogenetics class"),H={href:"https://brcwebportal.cos.ncsu.edu/thorne/bioinf2.html",target:"_blank",rel:"noopener noreferrer"},I=o("Jeffrey L. Thorne's grad-level Bioinformatics II class"),R={href:"https://vioshyvo.github.io/Bayesian_inference/",target:"_blank",rel:"noopener noreferrer"},G=o("Bayesian Inference notes with R examples by Ville Hyv\xF6nen & Topias Tolonen"),D={href:"https://stephens999.github.io/fiveMinuteStats/index.html",target:"_blank",rel:"noopener noreferrer"},F=o("Matthew Stephens's "),O=e("strong",null,"fiveMinuteStats",-1),z=o(": basic statistics concepts with practical R codes"),U={href:"http://bayesiancook.blogspot.com/",target:"_blank",rel:"noopener noreferrer"},W=o("Nicolas Lartillot's blog: The Bayesian kitchen"),q={href:"https://civil.colorado.edu/~balajir/CVEN6833/bayes-resources/RM-StatRethink-Bayes.pdf",target:"_blank",rel:"noopener noreferrer"},V=o("Book: Statistical Rethinking-A Bayesian Course with Examples in R and Stan by Richard McElreath"),K={href:"https://evolution.gs.washington.edu/gs541/2010/",target:"_blank",rel:"noopener noreferrer"},Y=o("Introduction to Computational Molecular Biology: Molecular Evolution at University of Washington in 2010"),Z=o(" Note: "),j={href:"https://evolution.gs.washington.edu/gs541/2010/20100420.mp3",target:"_blank",rel:"noopener noreferrer"},X=o("Dr. Felsenstein starts teaching Likelihood at 31'30"),Q={href:"https://eriqande.github.io/eca-bioinf-handbook/",target:"_blank",rel:"noopener noreferrer"},$=o("Eric C. Anderson's Handbook on Practical Computing and Bioinformatics for Conservation and Evolutionary Genomics"),ee={href:"https://github.com/eriqande/sisg-mcmc-opengl-computer-demos",target:"_blank",rel:"noopener noreferrer"},oe=o("Eric C. Anderson's MCMC simulations/visualizations demo"),te=o(" Note: "),ne={href:"https://www.youtube.com/watch?v=a8gjem86Uf4",target:"_blank",rel:"noopener noreferrer"},se=o("Eric C. Anderson's SISG MCMC OpenGL Demos tutorial on Youtube"),ae=e("h2",{id:"resources-related-to-hmc",tabindex:"-1"},[e("a",{class:"header-anchor",href:"#resources-related-to-hmc","aria-hidden":"true"},"#"),o(" Resources related to HMC")],-1),re={href:"https://arxiv.org/abs/1206.1901",target:"_blank",rel:"noopener noreferrer"},ie=o("Book Chapter: MCMC using Hamiltonian dynamics by Radford M. Neal"),le={href:"https://arxiv.org/pdf/1701.02434.pdf",target:"_blank",rel:"noopener noreferrer"},he=o("Book: A Conceptual Introduction to Hamiltonian Monte Carlo by Michael Betancourt"),ce=e("hr",{class:"footnotes-sep"},null,-1),de={class:"footnotes"},pe={class:"footnotes-list"},ue={id:"footnote1",class:"footnote-item"},fe={href:"http://treethinkers.org/update-must-read-papers-for-graduate-students/",target:"_blank",rel:"noopener noreferrer"},me=o("http://treethinkers.org/update-must-read-papers-for-graduate-students/"),_e=o(),ge=e("a",{href:"#footnote-ref1",class:"footnote-backref"},"\u21A9\uFE0E",-1),ye={id:"footnote2",class:"footnote-item"},be={href:"http://thomsonlab.org/people/bob-thomson/",target:"_blank",rel:"noopener noreferrer"},Me=o("http://thomsonlab.org/people/bob-thomson/"),ke=o(),ve=e("a",{href:"#footnote-ref2",class:"footnote-backref"},"\u21A9\uFE0E",-1),we={id:"footnote3",class:"footnote-item"},Be={href:"https://evolution.gs.washington.edu/sisg/2016/2016_SISG_19_5.pdf",target:"_blank",rel:"noopener noreferrer"},Se=o("https://evolution.gs.washington.edu/sisg/2016/2016_SISG_19_5.pdf"),Ce=o(),Ee=e("a",{href:"#footnote-ref3",class:"footnote-backref"},"\u21A9\uFE0E",-1),Ae={id:"footnote4",class:"footnote-item"},Pe={href:"https://evolution.gs.washington.edu/sisg/2016/2016_SISG_19_9.pdf",target:"_blank",rel:"noopener noreferrer"},xe=o("https://evolution.gs.washington.edu/sisg/2016/2016_SISG_19_9.pdf"),Le=o(),Je=e("a",{href:"#footnote-ref4",class:"footnote-backref"},"\u21A9\uFE0E",-1),Te={id:"footnote5",class:"footnote-item"},Ne={href:"https://brcwebportal.cos.ncsu.edu/thorne/ftp_docs/bioinf2/sampling2023.pdf",target:"_blank",rel:"noopener noreferrer"},He=o("https://brcwebportal.cos.ncsu.edu/thorne/ftp_docs/bioinf2/sampling2023.pdf"),Ie=o(),Re=e("a",{href:"#footnote-ref5",class:"footnote-backref"},"\u21A9\uFE0E",-1);function Ge(De,Fe){const t=l("ExternalLinkIcon");return a(),r("div",null,[c,e("ol",null,[e("li",null,[e("a",d,[p,n(t)])]),e("li",null,[e("a",u,[f,n(t)])]),e("li",null,[e("a",m,[_,n(t)])])]),g,e("p",null,[e("a",y,[b,n(t)])]),M,e("ol",null,[e("li",null,[e("a",k,[v,n(t)])]),e("li",null,[e("a",w,[B,n(t)])]),e("li",null,[e("a",S,[C,n(t)])]),e("li",null,[e("a",E,[A,n(t)])]),e("li",null,[e("a",P,[x,n(t)])]),e("li",null,[e("a",L,[J,n(t)])]),e("li",null,[e("a",T,[N,n(t)])]),e("li",null,[e("a",H,[I,n(t)])]),e("li",null,[e("a",R,[G,n(t)])]),e("li",null,[e("a",D,[F,O,z,n(t)])]),e("li",null,[e("a",U,[W,n(t)])]),e("li",null,[e("a",q,[V,n(t)])]),e("li",null,[e("a",K,[Y,n(t)]),Z,e("a",j,[X,n(t)])]),e("li",null,[e("a",Q,[$,n(t)])]),e("li",null,[e("a",ee,[oe,n(t)]),te,e("a",ne,[se,n(t)])])]),ae,e("ol",null,[e("li",null,[e("a",re,[ie,n(t)])]),e("li",null,[e("a",le,[he,n(t)])])]),ce,e("section",de,[e("ol",pe,[e("li",ue,[e("p",null,[e("a",fe,[me,n(t)]),_e,ge])]),e("li",ye,[e("p",null,[e("a",be,[Me,n(t)]),ke,ve])]),e("li",we,[e("p",null,[e("a",Be,[Se,n(t)]),Ce,Ee])]),e("li",Ae,[e("p",null,[e("a",Pe,[xe,n(t)]),Le,Je])]),e("li",Te,[e("p",null,[e("a",Ne,[He,n(t)]),Ie,Re])])])])])}const Ue=s(h,[["render",Ge],["__file","resources.html.vue"]]);export{Ue as default};
