import{_ as a}from"./_plugin-vue_export-helper.cdc0426e.js";import{o as s,c as r,b as e,d as n,f as i,e as o,r as l}from"./app.856165ed.js";const h={},c=i('<h2 id="must-read-papers-for-graduate-students" tabindex="-1"><a class="header-anchor" href="#must-read-papers-for-graduate-students" aria-hidden="true">#</a> Must read papers for graduate students</h2><p>The list was organized and posted on <sup class="footnote-ref"><a href="#footnote1">[1]</a><a class="footnote-anchor" id="footnote-ref1"></a></sup> by Bob Thomson <sup class="footnote-ref"><a href="#footnote2">[2]</a><a class="footnote-anchor" id="footnote-ref2"></a></sup> in 2015:</p><p>Bull, J. J., Huelsenbeck, J. P., Cunningham, C. W., Swofford, D. L., &amp; Waddel, P. J. (1993). Partitioning and combining data in phylogenetic analysis. Systematic Biology, 42(3), 384\u2013397.</p><p>Cavalli-Sforza, L. L., &amp; Edwards, a W. F. (1967). Phylogenetic analysis. Models and estimation procedures. The American Journal of Human Genetics, 19, 233\u2013257.</p><p>Edwards, S. V. (2009). Is a new and general theory of molecular systematics emerging? Evolution, 63, 1\u201319.</p><p>Felsenstein, J. (1973). Maximum likelihood and minimum-steps methods for estimating evolutionary trees from data on discrete characters. Systematic Biology, 22, 240\u2013249.</p><p>Felsenstein, J. (1978). Cases in which parsimony or compatibility methods will be positively misleading. Systematic Zoology, 27, 401\u2013410.</p><p>Felsenstein, J. (1981). Evolutionary trees from DNA sequences: a maximum likelihood approach. Journal of Molecular Evolution, 17, 368\u2013376.</p><p>Felsenstein, J. (1985). Confidence limits on phylogenies: an approach using the bootstrap. Evolution, 39, 783\u2013791.</p><p>Felsenstein, J. (1985). Phylogenies and the comparative method. American Naturalist, 125, 1\u201315.</p><p>Goldman, N. (1993). Statistical tests of models of DNA substitution. Journal of Molecular Evolution, 36, 182\u2013198.</p><p>Hillis, D. M., &amp; Bull, J. J. (1993). An Empirical Test of Bootstrapping as a Method for Assessing Confidence in Phylogenetic Analysis. Systematic Biology, 42, 182\u2013192.</p><p>Holder, M., &amp; Lewis, P. O. (2003). Phylogeny estimation: traditional and Bayesian approaches. Nature Reviews. Genetics, 4, 275\u2013284.</p><p>Kumar, S., Filipski, A. J., Battistuzzi, F. U., Kosakovsky Pond, S. L., &amp; Tamura, K. (2012). Statistics and truth in phylogenomics. Molecular Biology and Evolution, 29, 457\u2013472.</p><p>Maddison, W. P. (1997). Gene Trees in Species Trees. Systematic Biology, 46, 523\u2013536.</p><p>Pauling, L., &amp; Zuckerkandl, E. (1963). Chemical paleogenetics. Acta Chem. Scand, 17, S9 \u2013 S16.</p><p>Sullivan, J., &amp; Swofford, D. (1997). Are Guinea Pigs Rodents?? The Importance of Adequate Models in Molecular Phylogenetics. Journal of Mammalian Evolution, 4, 77\u201386.</p><h2 id="papers-related-to-hmm" tabindex="-1"><a class="header-anchor" href="#papers-related-to-hmm" aria-hidden="true">#</a> Papers related to HMM</h2><p>It is recommended by Joe Felsenstein <sup class="footnote-ref"><a href="#footnote3">[3]</a><a class="footnote-anchor" id="footnote-ref3"></a></sup> in 2016:</p><p>Yang, Z. 1994. Maximum-likelihood estimation of phylogeny from DNA sequences when substitution rates differ over sites. Molecular Biology and Evolution 10: 1396-1401. [Use of gamma distribution of rate variation in ML phylogenies]</p><p>Yang, Z. 1994. Maximum likelihood phylogenetic estimation from DNA sequences with variable rates over sites: approximate methods. Journal of Molecular Evolution 39: 306-314. [Approximating gamma distribution in ML phylogenies by an HMM]</p><p>Yang, Z. 1995. A space-time process model for the evolution of DNA sequences. Genetics 139: 993-1005. [Allowing for autocorrelated rates along the molecule using an HMM for ML phylogenies]</p><p>Felsenstein, J. and G. A. Churchill. 1996. A Hidden Markov Model approach to variation among sites in rate of evolution Molecular Biology and Evolution 13: 93-104. [HMM approach to evolutionary rate variation]</p><p>Thorne, J. L., N. Goldman, and D. T. Jones. 1996. Combining protein evolution and secondary structure. Molecular Biology and Evolution 13 666-673. [HMM for secondary structure of proteins, with phylogenies]</p><h2 id="papers-related-to-bayesian-model-comparison" tabindex="-1"><a class="header-anchor" href="#papers-related-to-bayesian-model-comparison" aria-hidden="true">#</a> Papers related to Bayesian model comparison</h2><p>It is recommended by Jeffrey Thorne <sup class="footnote-ref"><a href="#footnote4">[4]</a><a class="footnote-anchor" id="footnote-ref4"></a></sup> in 2016:</p><p>Posterior Predictive Inference in Phylogenetics: J.P. Bollback. 2002. Molecular Biology and Evolution. 19:1171-1180</p><p>Harmonic Mean and other techniques for estimating Bayes factors: Newton and Raftery. 1994. Journal of the Royal Statistical Society. Series B. 56(1):3-48.</p><div class="custom-container tip"><p class="custom-container-title">More reliable ways to approximate marginal likelihood</p><p>Thermodynamic Integration to Approximate Bayes Factors (adapted to molecular evolution data): Lartillot and Philippe. 2006. Syst. Biol. 55:195-207</p><p>Improving marginal likelihood estimation for Bayesian phylogenetic model selection. W. Xie, P.O. Lewis, Y. Fan, L. Kao, M-H Chen. 2011. Syst Biol. 60(2):150-160.</p><p>Choosing among partition models in Bayesian phylogenetics. Y. Fan, R. Wu, M-H Chen, L Kuo, P.O. Lewis. 2011. Mol. Biol. Evol. 28(1):523-532.</p></div><p>Markov chain Monte Carlo without likelihoods. P. Marjoram, J. Molitor, V. Plagnol, and S. Tavare. 2003. PNAS USA. 100(26): 15324-15328.</p><p>H. Jereys. The Theory of Probability (3e). Oxford (1961); p. 432</p><p>M.A. Beaumont, W. Zhang, D.J. Balding. Approximate Bayesian Computation in Population Genetics. 2002. Genetics 162:2025-2035.</p><h2 id="papers-related-to-markov-chain-monte-carlo" tabindex="-1"><a class="header-anchor" href="#papers-related-to-markov-chain-monte-carlo" aria-hidden="true">#</a> Papers related to Markov Chain Monte Carlo</h2><p>It is recommended by Jeffery Thorne <sup class="footnote-ref"><a href="#footnote5">[5]</a><a class="footnote-anchor" id="footnote-ref5"></a></sup></p><p>CARLIN, B.P., and T.A. LOUIS. 1996. Bayes and Empirical Bayes Methods for Data Analysis. Chapman and Hall, London.</p><p>GELMAN, A., J.B. CARLIN, H.S. STERN, and D.B. RUBIN. 1995. Bayesian Data Analysis. Chapman and Hall, London.</p><p>Hastings WK (1970) Monte Carlo sampling methods using Markov chains and their applications. Biometrika 57:97\u2013109</p><p>METROPOLIS, N., A.W. ROSENBLUTH, M.N. ROSENBLUTH, A.H. TELLER, and E. TELLER. 1953. Equations of state calculations by fast computing machines. J. Chem. Phys. 21: 1087\u20131092.</p><h2 id="other-awesome-recourses" tabindex="-1"><a class="header-anchor" href="#other-awesome-recourses" aria-hidden="true">#</a> Other awesome recourses</h2>',39),d={href:"https://felsenst.github.io/courses.html",target:"_blank",rel:"noopener noreferrer"},p=o("Joe Felsenstein's courses"),f={href:"https://evolution.gs.washington.edu/sisg/2016/",target:"_blank",rel:"noopener noreferrer"},u=o("2016 SISG Molecular Phylogenetics Module"),_={href:"http://www2.stat.duke.edu/~rcs46/bayes17.html",target:"_blank",rel:"noopener noreferrer"},m=o("Duke Bayesian Methods Class"),g={href:"http://treethinkers.org/2019-bodega-applied-phylogenetics-workshop-schedule/",target:"_blank",rel:"noopener noreferrer"},y=o("2019 Bodega Applied Phylogenetics Workshop"),b={href:"http://phyloseminar.org/",target:"_blank",rel:"noopener noreferrer"},M=o("Paul O. Lewis\u2019 Primer on Phylogenetics"),k={href:"https://plewis.github.io/phylogenetics2022/",target:"_blank",rel:"noopener noreferrer"},v=o("Paul O. Lewis' grad-level Phylogenetics class"),B={href:"https://brcwebportal.cos.ncsu.edu/thorne/bioinf2.html",target:"_blank",rel:"noopener noreferrer"},S=o("Jeffrey L. Thorne's grad-level Bioinformatics II class"),w={href:"https://vioshyvo.github.io/Bayesian_inference/",target:"_blank",rel:"noopener noreferrer"},P=o("Bayesian Inference notes with R examples"),A={href:"https://stephens999.github.io/fiveMinuteStats/index.html",target:"_blank",rel:"noopener noreferrer"},E=o("fiveMinuteStats"),C={href:"http://bayesiancook.blogspot.com/",target:"_blank",rel:"noopener noreferrer"},L=o("Blog: The Bayesian kitchen"),J={href:"https://civil.colorado.edu/~balajir/CVEN6833/bayes-resources/RM-StatRethink-Bayes.pdf",target:"_blank",rel:"noopener noreferrer"},x=o("Book: Statistical Rethinking-A Bayesian Course with Examples in R and Stan"),N={href:"https://evolution.gs.washington.edu/gs541/2010/",target:"_blank",rel:"noopener noreferrer"},T=o("Introduction to Computational Molecular Biology: Molecular Evolution"),I=o(" Note: "),H={href:"https://evolution.gs.washington.edu/gs541/2010/20100420.mp3",target:"_blank",rel:"noopener noreferrer"},R=o("Felsenstrin starts teaching Likelihood at 31'30"),G=e("h2",{id:"resources-related-to-hmc",tabindex:"-1"},[e("a",{class:"header-anchor",href:"#resources-related-to-hmc","aria-hidden":"true"},"#"),o(" Resources related to HMC")],-1),F={href:"https://arxiv.org/abs/1206.1901",target:"_blank",rel:"noopener noreferrer"},D=o("Book Chapter: MCMC using Hamiltonian dynamics by Radford M. Neal"),O={href:"https://arxiv.org/pdf/1701.02434.pdf",target:"_blank",rel:"noopener noreferrer"},W=o("Book: A Conceptual Introduction to Hamiltonian Monte Carlo by Michael Betancourt"),q=e("hr",{class:"footnotes-sep"},null,-1),U={class:"footnotes"},V={class:"footnotes-list"},K={id:"footnote1",class:"footnote-item"},Z={href:"http://treethinkers.org/update-must-read-papers-for-graduate-students/",target:"_blank",rel:"noopener noreferrer"},Y=o("http://treethinkers.org/update-must-read-papers-for-graduate-students/"),z=o(),j=e("a",{href:"#footnote-ref1",class:"footnote-backref"},"\u21A9\uFE0E",-1),X={id:"footnote2",class:"footnote-item"},Q={href:"http://thomsonlab.org/people/bob-thomson/",target:"_blank",rel:"noopener noreferrer"},$=o("http://thomsonlab.org/people/bob-thomson/"),ee=o(),oe=e("a",{href:"#footnote-ref2",class:"footnote-backref"},"\u21A9\uFE0E",-1),te={id:"footnote3",class:"footnote-item"},ne={href:"https://evolution.gs.washington.edu/sisg/2016/2016_SISG_19_5.pdf",target:"_blank",rel:"noopener noreferrer"},ae=o("https://evolution.gs.washington.edu/sisg/2016/2016_SISG_19_5.pdf"),se=o(),re=e("a",{href:"#footnote-ref3",class:"footnote-backref"},"\u21A9\uFE0E",-1),ie={id:"footnote4",class:"footnote-item"},le={href:"https://evolution.gs.washington.edu/sisg/2016/2016_SISG_19_9.pdf",target:"_blank",rel:"noopener noreferrer"},he=o("https://evolution.gs.washington.edu/sisg/2016/2016_SISG_19_9.pdf"),ce=o(),de=e("a",{href:"#footnote-ref4",class:"footnote-backref"},"\u21A9\uFE0E",-1),pe={id:"footnote5",class:"footnote-item"},fe={href:"https://brcwebportal.cos.ncsu.edu/thorne/ftp_docs/bioinf2/sampling2023.pdf",target:"_blank",rel:"noopener noreferrer"},ue=o("https://brcwebportal.cos.ncsu.edu/thorne/ftp_docs/bioinf2/sampling2023.pdf"),_e=o(),me=e("a",{href:"#footnote-ref5",class:"footnote-backref"},"\u21A9\uFE0E",-1);function ge(ye,be){const t=l("ExternalLinkIcon");return s(),r("div",null,[c,e("ol",null,[e("li",null,[e("a",d,[p,n(t)])]),e("li",null,[e("a",f,[u,n(t)])]),e("li",null,[e("a",_,[m,n(t)])]),e("li",null,[e("a",g,[y,n(t)])]),e("li",null,[e("a",b,[M,n(t)])]),e("li",null,[e("a",k,[v,n(t)])]),e("li",null,[e("a",B,[S,n(t)])]),e("li",null,[e("a",w,[P,n(t)])]),e("li",null,[e("a",A,[E,n(t)])]),e("li",null,[e("a",C,[L,n(t)])]),e("li",null,[e("a",J,[x,n(t)])]),e("li",null,[e("a",N,[T,n(t)]),I,e("a",H,[R,n(t)])])]),G,e("ol",null,[e("li",null,[e("a",F,[D,n(t)])]),e("li",null,[e("a",O,[W,n(t)])])]),q,e("section",U,[e("ol",V,[e("li",K,[e("p",null,[e("a",Z,[Y,n(t)]),z,j])]),e("li",X,[e("p",null,[e("a",Q,[$,n(t)]),ee,oe])]),e("li",te,[e("p",null,[e("a",ne,[ae,n(t)]),se,re])]),e("li",ie,[e("p",null,[e("a",le,[he,n(t)]),ce,de])]),e("li",pe,[e("p",null,[e("a",fe,[ue,n(t)]),_e,me])])])])])}const ve=a(h,[["render",ge],["__file","resources.html.vue"]]);export{ve as default};
