import{_ as t}from"./_plugin-vue_export-helper.cdc0426e.js";import{o,c as i,b as a,d as e,f as l,e as s,r}from"./app.a3980c23.js";const c={},p=l(`<h2 id="somaticsniper-v1-0-5-0" tabindex="-1"><a class="header-anchor" href="#somaticsniper-v1-0-5-0" aria-hidden="true">#</a> SomaticSniper (v1.0.5.0) <sup class="footnote-ref"><a href="#footnote1">[1]</a><a class="footnote-anchor" id="footnote-ref1"></a></sup></h2><h3 id="installation" tabindex="-1"><a class="header-anchor" href="#installation" aria-hidden="true">#</a> Installation</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token comment"># Build dependencies for Ubuntu</span>
<span class="token function">sudo</span> <span class="token function">apt-get</span> <span class="token function">install</span> build-essential git-core cmake zlib1g-dev libncurses-dev

<span class="token comment"># Clone the git repo</span>
<span class="token function">git</span> clone https://github.com/genome/somatic-sniper.git

<span class="token comment"># Build SomaticSniper</span>
<span class="token function">mkdir</span> somatic-sniper/build
<span class="token builtin class-name">cd</span> somatic-sniper/build
cmake <span class="token punctuation">..</span>/
<span class="token function">make</span> deps
<span class="token function">make</span> <span class="token parameter variable">-j</span>
<span class="token function">make</span> <span class="token builtin class-name">test</span>

<span class="token comment"># Add path to environment ~/.bashrc</span>
<span class="token builtin class-name">export</span> <span class="token assign-left variable"><span class="token environment constant">PATH</span></span><span class="token operator">=</span><span class="token environment constant">$PATH</span>:/PATH_TO/somatic-sniper/build/bin
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h3 id="usage" tabindex="-1"><a class="header-anchor" href="#usage" aria-hidden="true">#</a> Usage</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>bam-somaticsniper <span class="token punctuation">[</span>options<span class="token punctuation">]</span> <span class="token parameter variable">-f</span> <span class="token operator">&lt;</span>ref.fasta<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>tumor.bam<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>normal.bam<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>snv_output_file<span class="token operator">&gt;</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><p>Recommended setting:</p><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>bam-somaticsniper <span class="token parameter variable">-Q</span> <span class="token number">40</span> <span class="token parameter variable">-G</span> <span class="token parameter variable">-L</span> <span class="token parameter variable">-f</span> reference.fa tumor.bam normal.bam output.txt
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><h2 id="freebayes" tabindex="-1"><a class="header-anchor" href="#freebayes" aria-hidden="true">#</a> FreeBayes</h2><h3 id="installation-1" tabindex="-1"><a class="header-anchor" href="#installation-1" aria-hidden="true">#</a> Installation</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token function">wget</span> https://github.com/freebayes/freebayes/archive/refs/tags/v1.3.6.zip

<span class="token function">unzip</span> v1.3.6.zip
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h3 id="usage-1" tabindex="-1"><a class="header-anchor" href="#usage-1" aria-hidden="true">#</a> Usage</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>freebayes <span class="token parameter variable">-f</span> <span class="token punctuation">[</span>REFERENCE<span class="token punctuation">]</span> <span class="token punctuation">[</span>OPTIONS<span class="token punctuation">]</span> <span class="token punctuation">[</span>BAM FILES<span class="token punctuation">]</span> <span class="token operator">&gt;</span><span class="token punctuation">[</span>OUTPUT<span class="token punctuation">]</span>
freebayes <span class="token parameter variable">-f</span> ref.fa aln.bam <span class="token operator">&gt;</span>var.vcf

freebayes <span class="token parameter variable">-f</span> ref.fa --pooled-discrete --genotype-qualities tumor.bam normal.bam <span class="token operator">|</span> vcfsamplediff <span class="token parameter variable">-s</span> VT normal tumor -

</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><p>Genomic Research paper usage:</p><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>freebayes <span class="token parameter variable">-f</span> ref.fa  <span class="token parameter variable">-0</span> --pooled-discrete --genotype-qualities --min-coverage <span class="token number">20</span> tumor.bam normal.bam  <span class="token operator">&gt;</span> output.vcf
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><p>Then filter the vcf</p><h2 id="vcflib" tabindex="-1"><a class="header-anchor" href="#vcflib" aria-hidden="true">#</a> vcflib <sup class="footnote-ref"><a href="#footnote2">[2]</a><a class="footnote-anchor" id="footnote-ref2"></a></sup></h2><h3 id="installation-2" tabindex="-1"><a class="header-anchor" href="#installation-2" aria-hidden="true">#</a> Installation</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>conda <span class="token function">install</span> <span class="token parameter variable">-c</span> bioconda vcflib
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><h3 id="usage-2" tabindex="-1"><a class="header-anchor" href="#usage-2" aria-hidden="true">#</a> Usage</h3><p>Purpose: If multiple allelic primitives (gaps or mismatches) are specified in a single VCF record, split the record into multiple lines, but drop all INFO fields. Does not handle genotypes (yet). MNPs are split into multiple SNPs unless the -m flag is provided. Records generated by splits have th</p><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>vcfallelicprimitives <span class="token punctuation">[</span>options<span class="token punctuation">]</span> <span class="token punctuation">[</span>file<span class="token punctuation">]</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><h3 id="installation-3" tabindex="-1"><a class="header-anchor" href="#installation-3" aria-hidden="true">#</a> Installation</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token function">git</span> clone https://github.com/atks/vt.git
<span class="token builtin class-name">cd</span> vt
<span class="token function">git</span> submodule update <span class="token parameter variable">--init</span> <span class="token parameter variable">--recursive</span>
<span class="token function">make</span>
<span class="token comment"># Test the build</span>
<span class="token function">make</span> <span class="token builtin class-name">test</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h2 id="vt" tabindex="-1"><a class="header-anchor" href="#vt" aria-hidden="true">#</a> vt <sup class="footnote-ref"><a href="#footnote3">[3]</a><a class="footnote-anchor" id="footnote-ref3"></a></sup></h2><h3 id="usage-3" tabindex="-1"><a class="header-anchor" href="#usage-3" aria-hidden="true">#</a> Usage <sup class="footnote-ref"><a href="#footnote4">[4]</a><a class="footnote-anchor" id="footnote-ref4"></a></sup></h3><p>Purpose: Decompose biallelic block substitutions into their constituent SNVs</p><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token comment">#decomposes biallelic block substitutions and write out to decomposed_blocksub.vcf</span>
vt decompose_blocksub gatk.vcf <span class="token parameter variable">-o</span> decomposed_blocksub.vcf 
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div></div></div><h2 id="verscan" tabindex="-1"><a class="header-anchor" href="#verscan" aria-hidden="true">#</a> Verscan</h2><h3 id="installation-4" tabindex="-1"><a class="header-anchor" href="#installation-4" aria-hidden="true">#</a> Installation</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token function">wget</span> https://github.com/dkoboldt/varscan/releases/download/2.4.2/VarScan.v2.4.2.jar
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><h3 id="usage-4" tabindex="-1"><a class="header-anchor" href="#usage-4" aria-hidden="true">#</a> Usage</h3><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token function">java</span> <span class="token parameter variable">-jar</span> VarScan.jar  <span class="token punctuation">[</span>COMMAND<span class="token punctuation">]</span> <span class="token punctuation">[</span>OPTIONS<span class="token punctuation">]</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><h1 id="prepare-fasta-to-be-used-as-reference-genome" tabindex="-1"><a class="header-anchor" href="#prepare-fasta-to-be-used-as-reference-genome" aria-hidden="true">#</a> Prepare fasta to be used as reference genome</h1><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code><span class="token comment"># Create fasta.dict</span>
gatk CreateSequenceDictionary <span class="token parameter variable">-R</span> ref.fasta

<span class="token comment"># Create fasta.index</span>
samtools faidx ref.fasta
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h1 id="samtools" tabindex="-1"><a class="header-anchor" href="#samtools" aria-hidden="true">#</a> Samtools</h1><h2 id="samtools-view" tabindex="-1"><a class="header-anchor" href="#samtools-view" aria-hidden="true">#</a> <code>samtools view</code></h2><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>samtools view <span class="token punctuation">[</span>options<span class="token punctuation">]</span> <span class="token operator">&lt;</span>in.bam<span class="token operator">&gt;|</span><span class="token operator">&lt;</span>in.sam<span class="token operator">&gt;|</span><span class="token operator">&lt;</span>in.cram<span class="token operator">&gt;</span> <span class="token punctuation">[</span>region <span class="token punctuation">..</span>.<span class="token punctuation">]</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><p>Parameters</p><div class="language-javascript ext-js line-numbers-mode"><pre class="language-javascript"><code><span class="token operator">-</span><span class="token constant">S</span>           <span class="token function">Ignored</span> <span class="token punctuation">(</span>input format is auto<span class="token operator">-</span>detected<span class="token punctuation">)</span>
<span class="token operator">-</span>b<span class="token punctuation">,</span> <span class="token operator">--</span>bam                  Output <span class="token constant">BAM</span>
<span class="token operator">-</span>@<span class="token punctuation">,</span> <span class="token operator">--</span>threads <span class="token constant">INT</span>
               Number <span class="token keyword">of</span> additional threads to use <span class="token punctuation">[</span><span class="token number">0</span><span class="token punctuation">]</span>
<span class="token operator">-</span>o<span class="token punctuation">,</span> <span class="token operator">--</span>output <span class="token constant">FILE</span>          Write output to <span class="token constant">FILE</span> <span class="token punctuation">[</span>standard output<span class="token punctuation">]</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><h2 id="samtools-sort" tabindex="-1"><a class="header-anchor" href="#samtools-sort" aria-hidden="true">#</a> <code>samtools sort</code></h2><p>Usage:</p><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>samtools <span class="token function">sort</span> <span class="token punctuation">[</span>options<span class="token punctuation">..</span>.<span class="token punctuation">]</span> <span class="token punctuation">[</span>in.bam<span class="token punctuation">]</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><p>Parameters:</p><div class="language-javascript ext-js line-numbers-mode"><pre class="language-javascript"><code><span class="token operator">-</span>l <span class="token constant">INT</span>     Set compression level<span class="token punctuation">,</span> from <span class="token number">0</span> <span class="token punctuation">(</span>uncompressed<span class="token punctuation">)</span> to <span class="token number">9</span> <span class="token punctuation">(</span>best<span class="token punctuation">)</span> 
Set the desired compression level <span class="token keyword">for</span> the final output file<span class="token punctuation">,</span> ranging from <span class="token number">0</span> <span class="token punctuation">(</span>uncompressed<span class="token punctuation">)</span> or <span class="token number">1</span> <span class="token punctuation">(</span>fastest but minimal compression<span class="token punctuation">)</span> to <span class="token number">9</span> <span class="token punctuation">(</span>best compression but slowest to write<span class="token punctuation">)</span><span class="token punctuation">,</span> similarly to <span class="token function">gzip</span><span class="token punctuation">(</span><span class="token number">1</span><span class="token punctuation">)</span>&#39;s compression level setting<span class="token punctuation">.</span>

<span class="token operator">-</span>@ <span class="token constant">INT</span>
Set number <span class="token keyword">of</span> sorting and compression threads<span class="token punctuation">.</span> By <span class="token keyword">default</span><span class="token punctuation">,</span> operation is single<span class="token operator">-</span>threaded<span class="token punctuation">.</span>

<span class="token operator">-</span>o <span class="token constant">FILE</span>
Write the final sorted output to <span class="token constant">FILE</span><span class="token punctuation">,</span> rather than to standard output<span class="token punctuation">.</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div><div class="line-number"></div></div></div><hr><h2 id="samtools-fixmate" tabindex="-1"><a class="header-anchor" href="#samtools-fixmate" aria-hidden="true">#</a> <code>samtools fixmate</code></h2><p>Usage:</p><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>samtools fixmate <span class="token operator">&lt;</span>in.nameSrt.bam<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>out.nameSrt.bam<span class="token operator">&gt;</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><p>Parameters:</p><div class="language-javascript ext-js line-numbers-mode"><pre class="language-javascript"><code><span class="token operator">-</span>r           Remove unmapped reads and secondary alignments
<span class="token operator">-</span>m           Add mate score tag
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div><div class="line-number"></div></div></div><h2 id="samtools-markdup" tabindex="-1"><a class="header-anchor" href="#samtools-markdup" aria-hidden="true">#</a> <code>samtools markdup</code></h2><p>Usage:</p><div class="language-bash ext-sh line-numbers-mode"><pre class="language-bash"><code>samtools markdup <span class="token operator">&lt;</span>input.bam<span class="token operator">&gt;</span> <span class="token operator">&lt;</span>output.bam<span class="token operator">&gt;</span>
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><p>Parameters:</p><div class="language-javascript ext-js line-numbers-mode"><pre class="language-javascript"><code><span class="token operator">-</span>r                 Remove duplicate reads
</code></pre><div class="line-numbers" aria-hidden="true"><div class="line-number"></div></div></div><hr class="footnotes-sep">`,56),d={class:"footnotes"},u={class:"footnotes-list"},m={id:"footnote1",class:"footnote-item"},v={href:"https://github.com/genome/somatic-sniper",target:"_blank",rel:"noopener noreferrer"},h=s("https://github.com/genome/somatic-sniper"),b=s(),k=a("a",{href:"#footnote-ref1",class:"footnote-backref"},"\u21A9\uFE0E",-1),f={id:"footnote2",class:"footnote-item"},g={href:"https://github.com/vcflib/vcflib",target:"_blank",rel:"noopener noreferrer"},_=s("https://github.com/vcflib/vcflib"),x=s(),y=a("a",{href:"#footnote-ref2",class:"footnote-backref"},"\u21A9\uFE0E",-1),I={id:"footnote3",class:"footnote-item"},S={href:"https://github.com/atks/vt",target:"_blank",rel:"noopener noreferrer"},j=s("https://github.com/atks/vt"),w=s(),N=a("a",{href:"#footnote-ref3",class:"footnote-backref"},"\u21A9\uFE0E",-1),P={id:"footnote4",class:"footnote-item"},T={href:"https://genome.sph.umich.edu/wiki/Vt",target:"_blank",rel:"noopener noreferrer"},E=s("https://genome.sph.umich.edu/wiki/Vt"),V=s(),U=a("a",{href:"#footnote-ref4",class:"footnote-backref"},"\u21A9\uFE0E",-1);function B(F,O){const n=r("ExternalLinkIcon");return o(),i("div",null,[p,a("section",d,[a("ol",u,[a("li",m,[a("p",null,[a("a",v,[h,e(n)]),b,k])]),a("li",f,[a("p",null,[a("a",g,[_,e(n)]),x,y])]),a("li",I,[a("p",null,[a("a",S,[j,e(n)]),w,N])]),a("li",P,[a("p",null,[a("a",T,[E,e(n)]),V,U])])])])])}const L=t(c,[["render",B],["__file","variant_callers.html.vue"]]);export{L as default};
